cl-gopher.lisp

```diff
@@ -490,11 +490,11 @@
   #.(format nil "URI-FOR-GOPHER-LINE takes a GOPHER-LINE and returns~@
                  a string containing a gopher uri representing the~@
                  resource the line points to.")
-  (format nil "gopher://~a~:[:~a~;~*~]/~@[~*~c~a~]~@[~*%09~a~]"
+  (format nil "gopher://~a~:[:~a~;~*~]/~:[~2*~;~c~a~]~:[~*~;%09~a~]"
           (hostname gl)
           (eql (port gl) 70) (port gl)
           (not (or (null (selector gl))
                    (equal (selector gl) "")
                    (equal (selector gl) "/")))
           (type-character gl) (selector gl)
-          (not (uiop:emptyp (terms gl))) (quri:url-encode (terms gl))))
+          (not (uiop:emptyp (terms gl))) (quri:url-encode (or (terms gl) ""))))
```

Comment:
While we're here, maybe we can make this less confusing. How about we get rid of the `not`s since they just unnecessarily reorder the format conditionals:
```common-lisp
(defun uri-for-gopher-line (gl)
  #.(format nil "URI-FOR-GOPHER-LINE takes a GOPHER-LINE and returns~@
                 a string containing a gopher uri representing the~@
                 resource the line points to.")
  (format nil "gopher://~a~:[:~a~;~*~]/~:[~c~a~;~2*~]~:[%09~a~;~*~]"
          (hostname gl)
          (eql (port gl) 70) (port gl)
          (or (null (selector gl))
                   (equal (selector gl) "")
                   (equal (selector gl) "/"))
          (type-character gl) (selector gl)
          (uiop:emptyp (terms gl))
          (quri:url-encode (or (terms gl) ""))))
```